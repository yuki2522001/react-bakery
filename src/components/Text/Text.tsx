import React, { memo } from "react";

interface TitleProps {
  className?: string;
  text?: string;
}

const Text: React.FC<TitleProps> = memo(({ text, className }) => {
  return (
    <p data-testid="text" className={className}>
      {text}
    </p>
  );
});

export default Text;
