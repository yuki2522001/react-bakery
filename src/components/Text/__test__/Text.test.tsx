import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Text from "../Text";

describe("Price component", () => {
  test("should render price component", () => {
    const { getByTestId } = render(<Text text="This is new product" />);
    expect(getByTestId("text")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Text text="This is new product" />);
    expect(asFragment()).toMatchSnapshot();
  });
})