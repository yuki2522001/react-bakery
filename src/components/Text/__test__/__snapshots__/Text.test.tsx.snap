// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component matches snapshot 1`] = `
<DocumentFragment>
  <p
    data-testid="text"
  >
    This is new product
  </p>
</DocumentFragment>
`;
